src/iOS/Xamarin.iOS/Samples/Utility network/PerformValveIsolationTrace/PerformValveIsolationTrace.cs

```diff
@@ -0,0 +1,256 @@
+
+                // Create and load the utility network.
+                _utilityNetwork = await UtilityNetwork.CreateAsync(new Uri(FeatureServiceUrl));
+
+                // Create a map with layers in this utility network.
+                _myMapView.Map = new Map(Basemap.CreateStreetsNightVector());
+                _myMapView.Map.OperationalLayers.Add(new FeatureLayer(new Uri($"{FeatureServiceUrl}/{LineLayerId}")));
+                _myMapView.Map.OperationalLayers.Add(new FeatureLayer(new Uri($"{FeatureServiceUrl}/{DeviceLayerId}")));
+
+                // Get a trace configuration from a tier.
+                UtilityDomainNetwork domainNetwork = _utilityNetwork.Definition.GetDomainNetwork(DomainNetworkName) ?? throw new ArgumentException(DomainNetworkName);
+                UtilityTier tier = domainNetwork.GetTier(TierName) ?? throw new ArgumentException(TierName);
+                _configuration = tier.TraceConfiguration;
+
+                // Create a trace filter.
+                _configuration.Filter = new UtilityTraceFilter();
+
+                // Get a default starting location.
+                UtilityNetworkSource networkSource = _utilityNetwork.Definition.GetNetworkSource(NetworkSourceName) ?? throw new ArgumentException(NetworkSourceName);
+                UtilityAssetGroup assetGroup = networkSource.GetAssetGroup(AssetGroupName) ?? throw new ArgumentException(AssetGroupName);
+                UtilityAssetType assetType = assetGroup.GetAssetType(AssetTypeName) ?? throw new ArgumentException(AssetTypeName);
+                Guid globalId = Guid.Parse(GlobalId);
+                _startingLocation = _utilityNetwork.CreateElement(assetType, globalId);
+
+                // Create a graphics overlay.
+                GraphicsOverlay overlay = new GraphicsOverlay();
+                _myMapView.GraphicsOverlays.Add(overlay);
+                Symbol symbol = new SimpleMarkerSymbol(SimpleMarkerSymbolStyle.Cross, System.Drawing.Color.LimeGreen, 25d);
+
+                // Display starting location.
+                IEnumerable<ArcGISFeature> elementFeatures = await _utilityNetwork.GetFeaturesForElementsAsync(new List<UtilityElement> { _startingLocation });
+                MapPoint startingLocationGeometry = elementFeatures.FirstOrDefault().Geometry as MapPoint;
+                Graphic graphic = new Graphic(startingLocationGeometry, symbol);
+                overlay.Graphics.Add(graphic);
+
+                // Set the starting viewpoint.
+                await _myMapView.SetViewpointAsync(new Viewpoint(startingLocationGeometry, 3000));
+
+                // Build the choice list for categories populated with the `Name` property of each `UtilityCategory` in the `UtilityNetworkDefinition`.
+                //CategoryPicker.ItemsSource = _utilityNetwork.Definition.Categories.ToList();
+
+                // Enable the UI.
+                _categoryButton.Enabled = _traceButton.Enabled = true;
```

Comment:
The load time is slow enough that I thought it wasn't working at first. I recommend adding some sort of progress indicator or spinner while loading (applies to all platforms)

Reply:
Same for while the trace is happening